moto/ec2/models.py

```diff
@@ -2056,11 +2058,18 @@ def owner_id(self):
     def __eq__(self, other):
         if self.ip_protocol != other.ip_protocol:
             return False
-        if self.ip_ranges != other.ip_ranges:
+        ip_ranges = [item for item in self.ip_ranges if item not in other.ip_ranges]
```

Comment:
Should we use `set.symmetric_difference()` here to guarantee equality? 
Using the current comparison, the following scenario would still pass, if I'm not mistaken:
```
self.ip_ranges = [{"CidrIp": "0.0.0.0/0"}]
other.ip_ranges = [{"CidrIp": "1.1.1.1/0"}]
```